### PR TITLE
Add `missing_docs` check to newer crates

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
`mc-sgx-core-types` and `mc-sgx-tservice-types` crates will now error on
`missing_docs`, `missing_debug_implementations`, and `unsafe_code`

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #91** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/91" target="_blank">Graphite</a> 
    * **PR #86** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/86" target="_blank">Graphite</a> 
      * **PR #92** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/92" target="_blank">Graphite</a> 
        * **PR #103** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/103" target="_blank">Graphite</a> 
          * **PR #110** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/110" target="_blank">Graphite</a> 
            * **PR #111** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/111" target="_blank">Graphite</a> 
              * **PR #112** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/112" target="_blank">Graphite</a> 
                * **PR #113** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/113" target="_blank">Graphite</a> 
                  * **PR #115** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/115" target="_blank">Graphite</a> 
                    * **PR #116** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/116" target="_blank">Graphite</a> 
                      * **PR #117** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/117" target="_blank">Graphite</a> 
                        * **PR #118** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/118" target="_blank">Graphite</a> 
                          * **PR #119** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/119" target="_blank">Graphite</a> 
                            * **PR #121** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/121" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/121?utm_source=stack-comment).

File: core/types/src/attributes.rs [hunk 1]
```diff
@@ -23,33 +24,31 @@ impl Attributes {
         self
     }
 
-    /// Set the `transform` for the attributes
+    /// Set the extended features request mask (xfrm)
```
Comment (nick-mobilecoin) [inline core/types/src/attributes.rs:27]: renamed once I was able to track down a definition on [SGX 101](https://sgx101.gitbook.io/sgx101/sgx-bootstrap/enclave#sgx-enclave-attributes)

Comment (nick-mobilecoin) [reply core/types/src/attributes.rs:27]: Also not that this and `flags` should be using a bitflag style enum per the [developer reference](https://download.01.org/intel-sgx/sgx-linux/2.17.1/docs/Intel_SGX_Developer_Reference_Linux_2.17.1_Open_Source.pdf), but wanted to minimize this specific change

File: core/types/src/attributes.rs [hunk 2]
```diff
@@ -23,33 +24,31 @@ impl Attributes {
...
 new_type_accessors_impls! {
     MiscellaneousSelect, sgx_misc_select_t;
 }
 
-impl MiscellaneousSelect {
-    pub fn new(select: u32) -> Self {
-        Self(select)
-    }
-}
-
```
Comment (nick-mobilecoin) [inline core/types/src/attributes.rs:49]: Removed the `new` here, and elsewhere, as it was more than one way to do something with the existence of the `From` trait implementation.  Figured the `From` trait provides more utility

File: core/types/src/attestation_key.rs [hunk 3]
```diff
@@ -86,11 +86,13 @@ impl TryFrom<sgx_quote_sign_type_t> for QuoteSignatureKind {
     }
 }
 
+/// Attestation key from th quoting library.  contains the quoting enclaves ID
+/// and the attestation algorithm
 #[derive(Default, Debug, Clone, Hash, PartialEq, Eq)]
 #[repr(transparent)]
-pub struct QuoteLibAttestationKeyId(sgx_ql_att_key_id_t);
+pub struct AttestationKeyId(sgx_ql_att_key_id_t);
```
Comment (jcape) [inline core/types/src/attestation_key.rs:93]: 🙏🏻

File: core/types/src/quote.rs [hunk 4]
```diff
@@ -225,11 +225,27 @@ mod test {
...
+            slice::from_raw_parts(
+                &report as *const sgx_quote_t as *const u8,
+                mem::size_of::<sgx_quote_t>(),
+            )
+        };
+        let mut bytes: [u8; mem::size_of::<sgx_quote_t>()] = [0; mem::size_of::<sgx_quote_t>()];
+        bytes.copy_from_slice(alias_bytes);
+        bytes
+    }
+    fn base_quote_1() -> sgx_quote_t {
```
Comment (jcape) [inline core/types/src/quote.rs:244]:
```suggestion

    fn base_quote_1() -> sgx_quote_t {
```

**Review by jcape — APPROVED** [review]

**Review by awygle — APPROVED** [review]

**Comment by nick-mobilecoin** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/121?utm_source=stack-comment).


Job Reference: WM00OUpxmxIPJavmjHZu

**Pull request merged**